ceci/pipeline.py

```diff
@@ -124,14 +503,43 @@ def add_stage(self, stage_info):
 
         Parameters
         ----------
-        stage_info: dict
+        stage_info: dict or PipelineStage
             Configuration information for this stage. See docstring for info.
-
-
         """
-        sec = StageExecutionConfig(stage_info)
+        if isinstance(stage_info, PipelineStage):
+            sec = StageExecutionConfig.create(stage_info)
+        else:
+            sec = StageExecutionConfig(stage_info)
         self.stage_execution_config[sec.name] = sec
         self.stage_names.append(sec.name)
+        if sec.stage_obj is None:
+            return {}
+        return self.pipeline_files.insert_outputs(sec.stage_obj, '.')
```

Comment:
I think that is correct.  It's a bit of a frankenstein right now.  In interactive usage the Pipeline and the stages are primary location for the configuration (because if people update them, you want that to be latched, saved into provenance, written to snapshot, etc..) but in ballistic running the dict / yaml file is the primary location for the configuration.   

We could come back to this, but for now I think that this was the least disruptive way to get both cases to work.